### PR TITLE
Sidebar nav + conversation list layout overhaul

### PR DESCRIPTION
**Pull request by rbren**
Split out from `rb/megachange`. **Targets `rb/workspace-ux` because the new sidebar 'New Conversation' popover depends on `useResolvedWorkspaces` and the workspace-parents store introduced in that PR.** Rebase onto `main` once `rb/workspace-ux` lands.

This is the big interconnected layout refactor — I deliberately kept these commits together because they constantly trip over each other in `src/components/features/sidebar/`, `conversation-panel/`, and `routes/`.

- **Refactor sidebar nav and conversation list layout** (`90e53ca` equivalent): text-only nav (Conversations, Automations, Skills, Integrations, Settings with collapsible sub-menu); 220px column; `BackendSelector` pinned to bottom with a '+ Add backend' footer + modal; `/` redirects to `/conversations`; conversation-list pane is collapsible; chat title+model header above the chat pane; tabs above the right pane; `UserActions` / `AddBackendMenuItem` removed.
- **Add missing i18n keys** for Skills / Integrations / New buttons.
- **Move conversation list into the sidebar nav** (drop the separate pane component / route layout).
- **Sidebar new-conversation popover: flat workspace list.** The '+ New Conversation' trigger opens a flat list — leading 'No workspace' option plus one entry per resolved workspace — each clicking through immediately. Rename the trigger label from 'New' to 'New Conversation'.
- **Move LLM model name from conversation lists/title to the chat input** so the lists stay tighter and the model is visible exactly where the user is composing.
- **Conversation list polish:** clearer agent-status indicator, indent, workspace fallback, and 10s polling.
- **Invalidate (don't remove) conversations cache on create** so the list refreshes without flashing empty.
- **Old conversations:** restore the long-tail conversation list / context menu / confirm-delete modal behavior under the new layout.

_This PR was created by an AI agent (OpenHands) on behalf of @rbren as part of splitting the `rb/megachange` branch into reviewable PRs._

### CODE REVIEW

**Comment by rbren** [comment]
@Openhands address the code review feedback and fix merge conflicts

File: src/components/features/conversation-panel/conversation-card/conversation-card.tsx [hunk 1]
```diff
@@ -43,7 +44,8 @@ export function ConversationCard({
   executionStatus,
   contextMenuOpen = false,
   onContextMenuToggle,
-  llmModel,
+  isActive = false,
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-card/conversation-card.tsx:45]: Intentional change. This PR moves the model display out of the conversation card and into the chat composer/header, and I verified there are no remaining `llmModel` prop call sites on this branch. Keeping the tighter card layout; documenting the intent here and in the PR body. Addressed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/home/recent-conversations/conversation-status-dot.tsx [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
-const labelKeyFor = (status: V1ExecutionStatus | null | undefined): string => {
+type Visual = "check" | "working" | "paused" | "error" | "unknown";
+
+const FINISHED_GREEN = "#1FBD53";
+const WORKING_GREEN = "#1FBD53";
+const PAUSED_GRAY = "#A3A3A3";
+const ERROR_RED = "#FF684E";
+const UNKNOWN_GRAY = "#3C3C49";
+
+const visualFor = (status: V1ExecutionStatus | null | undefined): Visual => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/home/recent-conversations/conversation-status-dot.tsx]: Intentional UX change for this layout overhaul: the status indicator now differentiates finished / working / paused / error states more clearly in the tighter list. I kept that behavior and added focused regression coverage for the mapping and rendered variants in `__tests__/components/features/home/conversation-status-dot.test.tsx`. Addressed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/home/recent-conversations/conversation-status-dot.tsx [hunk 3]
```diff
@@ -30,24 +52,62 @@ export function ConversationStatusDot({
...
-        return "bg-[#A3A3A3]";
-      case V1ExecutionStatus.ERROR:
-      case V1ExecutionStatus.STUCK:
-        return "bg-[#A3A3A3]";
+  const visual = visualFor(executionStatus);
+  const label = t(labelKeyFor(visual));
+
+  // Fixed 10x10 box so all variants share the same footprint and avoid layout
+  // shift when the agent transitions between states.
+  const indicator = (() => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/home/recent-conversations/conversation-status-dot.tsx]: Added dedicated status-dot coverage in `__tests__/components/features/home/conversation-status-dot.test.tsx` for the rendered states and label mapping. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/new-conversation-button.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import { ManageWorkspacesModal } from "#/components/features/home/workspace-dropdown/manage-workspaces-modal";
+
+/**
+ * "+ New Conversation" trigger that opens an inline popover on top of the
+ * conversation list. The popover is a flat list: each entry (including a
+ * leading "No workspace" option) immediately starts a conversation when
+ * clicked. The sticky footer still exposes "+ Add Workspace" / "Manage
+ * Workspaces" entries.
+ */
+export function NewConversationButton() {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/new-conversation-button.tsx:24]: Added `__tests__/components/features/conversation-panel/new-conversation-button.test.tsx` covering popover toggle/dismissal, workspace launch, disabled state while creating, and the add/manage modal interactions. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation/conversation-name-context-menu.tsx [hunk 5]
```diff
@@ -56,13 +65,48 @@ export function ConversationNameContextMenu({
...
   const { t } = useTranslation("openhands");
   const { backend } = useActiveBackend();
   const { data: conversation } = useActiveConversation();
   const ref = useClickOutsideElement<HTMLUListElement>(onClose);
+
+  // When anchored, render via a portal with fixed positioning computed from
+  // the anchor's bounding rect. This avoids being clipped by ancestors with
+  // `overflow: hidden`.
+  const [portalStyle, setPortalStyle] = React.useState<React.CSSProperties>();
+  React.useLayoutEffect(() => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation/conversation-name-context-menu.tsx:82]: Added `__tests__/components/features/conversation/conversation-name-context-menu.test.tsx` covering inline vs portaled rendering, top/bottom positioning, listener cleanup, and the static-render null case. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-card/no-repository.tsx [hunk 6]
```diff
@@ -1,10 +1,46 @@
...
-export function NoRepository() {
+interface NoRepositoryProps {
+  workspaceWorkingDir?: string | null;
+}
+
+/**
+ * Returns the basename (top-level folder name) of a workspace directory path,
+ * tolerating both POSIX and Windows separators and trailing slashes.
+ */
+function basename(path: string): string {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-card/no-repository.tsx]: Moved the basename logic into `src/utils/path-utils.ts` as `getPathBasename()` and added cross-platform coverage in `__tests__/utils/path-utils.test.ts`. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-panel.tsx [hunk 7]
```diff
@@ -61,17 +89,28 @@ export function ConversationPanel({ onClose }: ConversationPanelProps) {
   // Flatten all pages into a single array of conversations (V1 uses 'items' instead of 'results')
   const conversations = data?.pages.flatMap((page) => page.items) ?? [];
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-panel.tsx]: Updated `ConversationPanel` to memoize the flattened `conversations` array directly and partition from that memoized value, so the extra eslint suppression is gone. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-panel.tsx [hunk 8]
```diff
@@ -121,87 +160,163 @@ export function ConversationPanel({ onClose }: ConversationPanelProps) {
     }
   };
 
+  const handleConfirmDeleteOlder = () => {
+    const idsToDelete = olderConversations.map((c) => c.id);
+    const willClearActive =
+      currentConversationId !== null &&
+      idsToDelete.includes(currentConversationId);
+    for (const id of idsToDelete) {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-panel.tsx]: Updated bulk-delete handling to use `mutateAsync` + `Promise.allSettled`, then only navigate away if the active conversation was actually deleted. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-panel.tsx [hunk 9]
```diff
@@ -105,7 +144,7 @@ export function ConversationPanel({ onClose }: ConversationPanelProps) {
         {
           onSuccess: () => {
             if (selectedConversationId === currentConversationId) {
-              navigate("/");
+              navigate("/conversations");
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-panel.tsx:152]: Kept `navigate('/conversations')` intentionally. Under the new layout `/` immediately redirects to `/conversations`, so navigating directly to the canonical route avoids relying on an extra redirect hop when the active conversation disappears. Documented here; behavior remains intentional in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/backends/backend-selector.tsx [hunk 10]
```diff
@@ -78,6 +85,7 @@ export function BackendSelector() {
     useSwitchCloudOrganization();
   const navigate = useNavigate();
   const conversationMatch = useMatch("/conversations/:conversationId");
+  const [addBackendModalOpen, setAddBackendModalOpen] = React.useState(false);
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx:100]: Kept the add-backend modal state local to `BackendSelector` intentionally because the footer/trigger now live inside that component. This keeps the selector self-contained in the sidebar footer. I also added regression coverage for the footer + modal interaction in `__tests__/components/backends/backend-selector.test.tsx`. Addressed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/backends/backend-selector.tsx [hunk 11]
```diff
@@ -135,11 +143,26 @@ export function BackendSelector() {
       });
   }, [active, cloudOrgs, currentUserIds, setActive, switchOrg]);
 
+  const addBackendFooter = (
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx:203]: Added coverage for the selector footer and add-backend modal open/close flow in `__tests__/components/backends/backend-selector.test.tsx`, and also fixed the actual footer interaction so the dropdown doesn't collapse before the modal opens. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation/conversation-name-context-menu.tsx [hunk 12]
```diff
@@ -56,13 +65,48 @@ export function ConversationNameContextMenu({
...
+      } else {
+        style.top = rect.bottom + gap;
+      }
+      style.left = rect.left;
+      setPortalStyle(style);
+    };
+
+    updatePosition();
+    window.addEventListener("resize", updatePosition);
+    window.addEventListener("scroll", updatePosition, true);
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation/conversation-name-context-menu.tsx:105]: I kept the current listener behavior for now. The anchored menu only attaches the listeners while it is mounted/open, and the new tests cover both positioning and cleanup. I don't want to add throttling complexity until we see a concrete issue here. Decision documented in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/home/recent-conversations/conversation-status-dot.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/home/recent-conversations/conversation-status-dot.tsx]: Simplified the indicator rendering by replacing the IIFE with a dedicated `renderIndicator()` helper. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/home/recent-conversations/conversation-status-dot.tsx [hunk 13]
```diff
@@ -1,25 +1,47 @@
...
 import { StyledTooltip } from "#/components/shared/buttons/styled-tooltip";
 
 interface ConversationStatusDotProps {
   executionStatus: V1ExecutionStatus | null | undefined;
 }
 
-const labelKeyFor = (status: V1ExecutionStatus | null | undefined): string => {
+type Visual = "check" | "working" | "paused" | "error" | "unknown";
+
+const FINISHED_GREEN = "#1FBD53";
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/home/recent-conversations/conversation-status-dot.tsx]: Collapsed the duplicate green constants into a single `SUCCESS_GREEN` constant while keeping the intentional visuals the same. Fixed in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-panel.tsx [hunk 14]
```diff
@@ -21,13 +20,38 @@ import { StartTaskCard } from "./start-task-card/start-task-card";
...
 interface ConversationPanelProps {
-  onClose: () => void;
+  onClose?: () => void;
 }
 
+const noop = () => {};
+
+const ONE_HOUR_MS = 60 * 60 * 1000;
+
+const partitionByCutoff = <T extends { updated_at: string }>(
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-panel.tsx:33]: I kept the current one-pass partition helper. With the conversations list memoized upstream now, this stays simple and is fine for current scale; if list size becomes a real issue later, I'd prefer to address that with virtualization/pagination rather than pre-optimizing here. Decision documented in 9467d59.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

**Comment by rbren** [comment]
@openhands the "add backend" modal is not opening. Make it open the existing modal. also add a "manage backends" modal which allows you to delete existing backends

**Comment by rbren** [comment]
@openhands address the PR feedback

File: public/mockServiceWorker.js [hunk 15]
```diff
@@ -12,104 +12,104 @@ const INTEGRITY_CHECKSUM = '4db4a41e972cec1b64cc569c66952d82'
 const IS_MOCKED_RESPONSE = Symbol('isMockedResponse')
 const activeClientIds = new Set()
 
-addEventListener("install", function () {
-  self.skipWaiting();
-});
+addEventListener('install', function () {
```
[the remark this answers is not in the document]
Comment (rbren) [reply public/mockServiceWorker.js:15]: Reverted `public/mockServiceWorker.js` back to the generated MSW copy in 91713b5, so the formatting-only churn is gone.\n\n_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/backends/backend-selector.tsx [hunk 16]
```diff
@@ -139,63 +152,92 @@ export function BackendSelector() {
...
+
+  const addBackendFooter = (
+    <div className="flex flex-col gap-1">
+      <button
+        type="button"
+        data-testid="add-backend-menu-item"
+        onMouseDown={(event) => {
+          event.preventDefault();
+          event.stopPropagation();
+          openAddBackendModal();
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx]: Fixed in 91713b5. The footer buttons now only prevent menu teardown on `onMouseDown`; the modal open stays on `onClick`, matching the workspace dropdown pattern.\n\n_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/backends/backend-selector.tsx [hunk 17]
```diff
@@ -139,63 +152,92 @@ export function BackendSelector() {
...
+
+          triggerEnvironmentSwitch(item.label);
+          await new Promise<void>((resolve) => {
+            setTimeout(resolve, ENVIRONMENT_SWITCH_SETACTIVE_DELAY_MS);
+          });
+
+          if (orgId && target?.kind === "cloud") {
+            try {
+              await switchOrg({ orgId, backend: target });
+            } catch {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx]: Fixed in 91713b5. Failed cloud-org switches now surface an error toast, and the regression test covers the failure path without persisting the cloud selection.\n\n_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: AGENTS.md [hunk 18]
```diff
@@ -139,6 +140,7 @@
...
   - `FolderBrowserModal`'s "Use this folder" button adds **only the currently navigated directory** as a single workspace (named by its basename). It no longer iterates `subdirs` and adds each child as a separate workspace.
   - The `WorkspaceDropdown` sticky footer now exposes both "+ Add Workspace" (opens the folder browser) and "Manage Workspaces" (opens `ManageWorkspacesModal`, which lets users remove individual workspaces via `useWorkspacesStore.removeWorkspace`). The Manage button is hidden when there are no workspaces yet.
+  - The sidebar "+ New Conversation" trigger (`NewConversationButton` in `src/components/features/conversation-panel/`) opens a popover that is a **flat list**, not the home-screen combobox: a leading "No workspace" entry plus one entry per stored workspace, each clicking through to `useCreateConversation` immediately (no separate Launch button). The same sticky "+ Add Workspace" / "Manage Workspaces" footer is reused from the dropdown component, but the embedded `WorkspaceDropdown` itself is gone from this trigger.
```
[the remark this answers is not in the document]
Comment (rbren) [reply AGENTS.md]: Fixed in 91713b5. I updated the AGENTS note to describe the actual pattern (`onMouseDown` keeps the menu mounted, `onClick` opens the modal) and clarified that `NewConversationButton` mirrors the footer actions locally instead of literally reusing `WorkspaceDropdown`.\n\n_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/chat/components/chat-input-model.tsx [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  return (
+    <span
+      className="text-sm font-normal leading-5 text-[#A3A3A3] flex items-center gap-1 whitespace-nowrap"
+      title={conversation.llm_model}
+      data-testid="chat-input-llm-model"
+    >
+      <CircuitIcon width={14} height={14} className="shrink-0" />
+      <span>{conversation.llm_model}</span>
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/chat/components/chat-input-model.tsx]: Fixed in 91713b5. The model pill now constrains its width and truncates long model names while keeping the full value in the title tooltip.\n\n_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-panel.tsx [hunk 20]
```diff
@@ -21,13 +20,38 @@ import { StartTaskCard } from "./start-task-card/start-task-card";
...
+  items: readonly T[],
+): { recent: T[]; older: T[] } => {
+  const cutoff = Date.now() - ONE_HOUR_MS;
+  const recent: T[] = [];
+  const older: T[] = [];
+  for (const item of items) {
+    const updatedAt = item.updated_at ? Date.parse(item.updated_at) : NaN;
+    if (Number.isFinite(updatedAt) && updatedAt < cutoff) {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-panel.tsx:46]: Fixed in 91713b5. I added an inline note explaining that invalid or missing timestamps intentionally stay in the recent bucket so they do not get hidden behind the older-conversations toggle.\n\n_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

**Comment by rbren** [comment]
@openhands fix the merge conflicts

**Comment by rbren** [comment]
@Openhands address the latest code review feedback

**Comment by rbren** [comment]
Addressed the latest review feedback in `212708f`:

- fixed the remaining low-risk UI/code-quality items called out in review (`ConfirmDeleteModal` now treats `null` like an absent custom description, `NewConversationButton` uses a viewport-aware popover max height, the stale `lastUpdatedAt`/spacing comments are cleaned up, and `ConversationPanel` now documents the moving cutoff and memoizes the card renderer path)
- added regression coverage for invalid/missing conversation timestamps + older/load-more behavior, failed new-conversation creation, and `stripWorkspacePrefix()`

Validation:
- `npm run make-i18n`
- `./node_modules/.bin/vitest run __tests__/components/features/conversation-panel/confirm-delete-modal.test.tsx __tests__/components/features/conversation-panel/conversation-panel.test.tsx __tests__/components/features/conversation-panel/new-conversation-button.test.tsx __tests__/utils/path-utils.test.ts`
- `npm run typecheck`
- `npm run build`

_This comment was created by an AI agent (OpenHands) on behalf of @rbren._

**Comment by rbren** [comment]
@openhands address the code review comments and resolve them once you're done

File: src/components/features/backends/backend-selector.tsx [hunk 21]
```diff
@@ -140,68 +156,97 @@ export function BackendSelector() {
...
+          await new Promise<void>((resolve) => {
+            setTimeout(resolve, ENVIRONMENT_SWITCH_SETACTIVE_DELAY_MS);
+          });
+
+          if (orgId && target?.kind === "cloud") {
+            try {
+              await switchOrg({ orgId, backend: target });
+            } catch (error) {
+              displayErrorToast(
+                retrieveAxiosErrorMessage(error as AxiosError) ||
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx]: Handled this by switching the catch path to `axios.isAxiosError(error)` and only showing the retrieved Axios message for actual request failures; unexpected errors are rethrown instead of being cast. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/backends/backend-selector.tsx [hunk 22]
```diff
@@ -140,68 +156,97 @@ export function BackendSelector() {
...
+
+          triggerEnvironmentSwitch(item.label);
+          await new Promise<void>((resolve) => {
+            setTimeout(resolve, ENVIRONMENT_SWITCH_SETACTIVE_DELAY_MS);
+          });
+
+          if (orgId && target?.kind === "cloud") {
+            try {
+              await switchOrg({ orgId, backend: target });
+            } catch (error) {
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx:255]: Addressed in the same error-handling pass: the org-switch catch block now dismisses the overlay, handles Axios failures explicitly, and rethrows non-Axios errors so programming errors are not silently swallowed. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/backends/backend-selector.tsx [hunk 23]
```diff
@@ -140,68 +156,97 @@ export function BackendSelector() {
...
+          activeOption ?? { value: activeValue, label: bundledLabel }
         }
+        footer={addBackendFooter}
+        openUpward={openUpward}
+        onChange={async (item) => {
+          if (!item || item.value === activeValue) return;
+          const { backendId, orgId } = parseOptionValue(item.value);
+          const target = backends.find((b) => b.id === backendId);
+
+          triggerEnvironmentSwitch(item.label);
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/backends/backend-selector.tsx:247]: Implemented the early-dismiss path on org-switch failure via `dismissEnvironmentSwitch()` and added regression coverage in `backend-selector.test.tsx` / `environment-switch-overlay.test.tsx`. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: __tests__/components/backends/backend-selector.test.tsx [hunk 24]
```diff
@@ -427,6 +439,120 @@ describe("BackendSelector", () => {
...
+    await user.click(screen.getByTestId("add-backend-menu-item"));
+    expect(await screen.findByTestId("add-backend-modal")).toBeInTheDocument();
+
+    await user.click(screen.getByTestId("add-backend-cancel"));
+    await waitFor(() => {
+      expect(screen.queryByTestId("add-backend-modal")).not.toBeInTheDocument();
+    });
+  });
+
+  it("opens the manage backends modal and removes a backend", async () => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply __tests__/components/backends/backend-selector.test.tsx:613]: Added the missing manage-backends regression test for removing the currently active backend. It now verifies that the active persisted selection is cleared/falls back to the bundled default UI state after removal. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation-panel/conversation-panel.tsx [hunk 25]
```diff
@@ -121,87 +164,178 @@ export function ConversationPanel({ onClose }: ConversationPanelProps) {
     }
   };
 
+  const handleConfirmDeleteOlder = async () => {
+    const idsToDelete = olderConversations.map((c) => c.id);
+    const results = await Promise.allSettled(
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation-panel/conversation-panel.tsx:170]: Updated bulk delete to count rejected deletions and surface an error toast with the failure count, instead of failing silently. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: __tests__/components/features/conversation-panel/conversation-panel.test.tsx [hunk 26]
```diff
@@ -273,6 +279,78 @@ describe("ConversationPanel", () => {
...
+      new Date(now - minutes * 60 * 1000).toISOString();
+
+    const user = userEvent.setup();
+    const searchConversationsSpy = vi.spyOn(
+      AgentServerConversationService,
+      "searchConversations",
+    );
+    searchConversationsSpy.mockReset();
+    searchConversationsSpy
+      .mockResolvedValueOnce({
```
[the remark this answers is not in the document]
Comment (rbren) [reply __tests__/components/features/conversation-panel/conversation-panel.test.tsx:309]: Added partial-failure coverage for bulk delete: one case where the active older conversation succeeds and navigation still happens, and one where the active ID fails and navigation does not happen. Both assert the failure toast as well. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: __tests__/components/features/conversation-panel/conversation-card.test.tsx [hunk 27]
```diff
@@ -442,31 +442,7 @@ describe("ConversationCard", () => {
...
-    expect(model.querySelector("svg")).toBeInTheDocument();
-
-    // Verify truncation structure: text is wrapped in a span with truncate class
-    const textSpan = model.querySelector("span.truncate");
-    expect(textSpan).toBeInTheDocument();
-    expect(textSpan).toHaveTextContent("anthropic/claude-sonnet-4-20250514");
-  });
-
-  it("should not render the llm model when not provided", () => {
+  it("should not render the llm model in the conversation card", () => {
```
[the remark this answers is not in the document]
Comment (rbren) [reply __tests__/components/features/conversation-panel/conversation-card.test.tsx:492]: Added `workspaceWorkingDir` regression coverage in `conversation-card.test.tsx` for POSIX basename extraction, Windows path handling, and empty-path fallback to the no-repository label. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/chat/components/chat-input-actions.tsx [hunk 28]
```diff
@@ -36,6 +37,7 @@ export function ChatInputActions({ disabled }: ChatInputActionsProps) {
       <div className="flex items-center gap-1">
         <div className="flex items-center gap-4">
           <Tools />
+          <ChatInputModel />
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/chat/components/chat-input-actions.tsx:40]: Added parent-level `ChatInputActions` coverage for the relocated model display, including both `llm_model` present and absent cases via `useActiveConversation`. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: AGENTS.md [hunk 29]
```diff
@@ -144,6 +145,16 @@
...
+  - `useResolvedWorkspaces()` now returns `isLoading` / `isError` for parent-directory scans; `WorkspaceSelectionForm` should surface that state (status text and disabling the empty dropdown while parent results are still loading) instead of assuming the merged list is immediately ready.
+  - `ManageWorkspacesModal` should require a confirmation step before removing either a saved workspace or a workspace parent; parent removals should mention the child-workspace impact, and tests should assert both the confirmation flow and that removing the selected workspace clears the launch selection.
+  - In `useWorkspacesStore`, keep `clearWorkspaces()` scoped to literal workspaces only; use explicit helpers like `clearWorkspaceParents()` / `clearAll()` for broader resets so future callers do not accidentally wipe parent registrations.
```
[the remark this answers is not in the document]
Comment (rbren) [reply AGENTS.md:153]: Updated `AGENTS.md` to remove the inaccurate duplicate RepoConnector branching note so the repo memory matches the current implementation (workspace-only launcher). Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: AGENTS.md [hunk 30]
```diff
@@ -144,6 +145,16 @@
   - In `useWorkspacesStore`, keep `clearWorkspaces()` scoped to literal workspaces only; use explicit helpers like `clearWorkspaceParents()` / `clearAll()` for broader resets so future callers do not accidentally wipe parent registrations.
 
 
+- Home page workspace UX (agent-server backend):
```
[the remark this answers is not in the document]
Comment (rbren) [reply AGENTS.md:146]: Merged the duplicated home-page workspace UX guidance in `AGENTS.md` while keeping the current canonical section. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: __tests__/utils/path-utils.test.ts [hunk 31]
```diff
@@ -0,0 +1,52 @@
...
+      "src/file.py",
+    );
+    expect(
+      stripWorkspacePrefix("/workspace/my-project/components/Button.tsx"),
+    ).toBe("components/Button.tsx");
+  });
+
+  it("leaves non-workspace or incomplete paths unchanged", () => {
+    expect(stripWorkspacePrefix("/workspace")).toBe("/workspace");
+    expect(stripWorkspacePrefix("/workspace/repo")).toBe("/workspace/repo");
```
[the remark this answers is not in the document]
Comment (rbren) [reply __tests__/utils/path-utils.test.ts:52]: Added the trailing-slash edge case for `stripWorkspacePrefix("/workspace/repo/")` in `path-utils.test.ts`; it now asserts the empty-string result explicitly. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

File: src/components/features/conversation/conversation-name-context-menu.tsx [hunk 32]
```diff
@@ -56,13 +65,48 @@ export function ConversationNameContextMenu({
...
+    return () => {
+      window.removeEventListener("resize", updatePosition);
+      window.removeEventListener("scroll", updatePosition, true);
+    };
+  }, [anchorRef, position]);
```
[the remark this answers is not in the document]
Comment (rbren) [reply src/components/features/conversation/conversation-name-context-menu.tsx]: Addressed by keying the layout effect off `anchorElement` (`anchorRef?.current`) rather than the stable ref object, and added a regression test that swaps the anchor element between renders and verifies the menu repositions correctly. Fixed in e85e248.

_This reply was created by an AI agent (OpenHands) on behalf of @rbren._

**Comment by rbren** [comment]
@openhands address the code review comments and resolve them once you're done

**Comment by rbren** [comment]
Addressed the latest PR review follow-ups in `8b3467c` and pushed them to `rb/sidebar-and-conversation-list-layout`.

What changed:
- hardened `BackendSelector` so unexpected org-switch errors surface a generic toast instead of rethrowing from the dropdown handler, and malformed cloud/null-org self-heal falls back to the bundled backend if `/switch` fails
- added Escape-key dismissal for the sidebar `NewConversationButton` popover
- tightened regression coverage for backend switching, new-conversation popover behavior, and bulk-delete call ordering

Verification:
- `npm test -- --run __tests__/components/backends/backend-selector.test.tsx __tests__/components/features/conversation-panel/new-conversation-button.test.tsx __tests__/components/features/conversation-panel/conversation-panel.test.tsx`
- `npm run typecheck`
- `npm run build`

I also checked the PR review threads after pushing: the latest review was summary-only, and there are currently no unresolved review threads left to resolve.

_This comment was created by an AI agent (OpenHands) on behalf of @rbren._

**Comment by rbren** [comment]
@openhands fix the merge conflicts

**Comment by rbren** [comment]
@openhands fix the failing tests
